kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -623,8 +648,7 @@ KeysVector DBAdapter::lastReachableBlockKeyDeletes(BlockId blockId) const {
   // Clear the stale index for the last version.
   add(keysToDelete, staleIndexKeysForVersion(blockNode.stateRootVersion));
 
-  // TODO(DD): Add to keysToDelete the NonProvable keys and NonProvableStale(if any).
-  // key_type|NonProvableStale|stale_since_version|
+  add(keysToDelete, staleIndexKeysForBlock(blockId));
```

Comment:
If I am not mistaken, we are not actually deleting the non-provable keys. Instead, we are only deleting the stale indices for non-provable keys. I think that's also related to the fact we don't put non-provable keys in the block node and we might reconsider doing that as I mentioned above.